api/src/main/java/org/openmrs/api/db/hibernate/HibernateUserDAO.java

```diff
@@ -360,8 +361,10 @@ public boolean isSecretAnswer(User u, String answer) throws DAOException {
 			return false;
 		}
 		
-		String answerOnRecord = getLoginCredential(u).getSecretAnswer();
-		return (answer.equals(answerOnRecord));
+		LoginCredential credentials = getLoginCredential(u);
+		String answerOnRecord = credentials.getSecretAnswer();
+		String hashedAnswer = Security.encodeString(answer + credentials.getSalt());

```

Comment:
I would suggest `answer.toLowerCase()` to make secret answer validation case insensitive.